builder/builder.go

```diff
@@ -443,13 +432,6 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 		return nil, errors.Wrapf(err, "build main")
 	}
 	sp.printCurrentSuccess()
-	if outDir == "" {
-		var err error
-		outDir, err = b.tempEarthlyOutDir()
-		if err != nil {
-			return nil, err
-		}
-	}
```

Comment:
The removal of this section is the key part of the PR, right?

Reply:
This is the section that was giving the most grief, as it was called even when we were running with `opt.NoOutput`.